packages/cdk/lib/roles/policies/toil-batch-policy.ts

```diff
@@ -0,0 +1,33 @@
+import { PolicyDocument, PolicyStatement, Effect } from "aws-cdk-lib/aws-iam";
+
+export interface ToilBatchPolicyProps {
+  jobQueueArn: string;
+  // This is actually a pattern that matches all ARNs for potentially relevant
+  // definitions, since Toil makes its own definitions.
+  toilJobArn: string;
+}
+
+export class ToilBatchPolicy extends PolicyDocument {
+  constructor(props: ToilBatchPolicyProps) {
+    super({
+      assignSids: true,
+      statements: [
+        new PolicyStatement({
+          effect: Effect.ALLOW,
+          actions: ["batch:DescribeJobDefinitions", "batch:ListJobs", "batch:DescribeJobs", "batch:DescribeJobQueues", "batch:DescribeComputeEnvironments"],
+          resources: ["*"],
```

Comment:
I think some of these can have a more limited scope. For example, you know the job queue so possibly `batch:DescribeJobQueues` can be limited to that arn. You may also be able to limit `batch:DescribeJobDefinitions` to the `toilJobDeinitionArnPattern`. You may have to split the policies to do this but in general the number of actions with `resources: ["*"]` should be as small as possible

Reply:
This is just a copy-paste of the policy that Cromwell is using:

https://github.com/aws/amazon-genomics-cli/blob/8bda67492b3cf143ffd5a0a8931bbcdf3e04992d/packages/cdk/lib/roles/policies/cromwell-batch-policy.ts#L13-L17

Testing any change here is going to take something like an hour of turnaround time to tear down and redeploy the stack, so I'm very reluctant to undertake myself the project of determining what the minimum permission set to use Batch actually is.

I think instead of copy-pasting the code, I will try and base this off the Cromwell policy more directly, so that when the Cromwell permission set gets tightened, Toil will also be fixed.